cinema/tests/test_movie_api.py

```diff
@@ -157,3 +158,179 @@
         res = self.client.get(MOVIE_SESSION_URL)
 
         self.assertIn("movie_image", res.data[0].keys())
+
+
+class UnauthenticatedCinemaAPITests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_unauthenticated_user_cannot_access_movie_list(self):
+        res = self.client.get(MOVIE_URL)
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class AuthenticatedCinemaAPITests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="test@test.test",
+            password="test_password"
+        )
+        self.client.force_authenticate(self.user)
+
+    def test_movies_list(self):
+        sample_movie()
+        movie_with_genres = sample_movie()
+        movie_with_actors = sample_movie()
+
+        genre = sample_genre()
+        actor = sample_actor()
+
+        movie_with_genres.genres.add(genre)
+        movie_with_actors.actors.add(actor)
+
+
+        response= self.client.get(MOVIE_URL)
+        movies = Movie.objects.all()
+        serializer = MovieListSerializer(movies, many=True)
+
+        self.assertEqual(response.data, serializer.data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_filter_movies_by_title(self):
+        movie_1 = sample_movie()
+        movie_2 = sample_movie(title="Test_1")
+        movie_3 = sample_movie(title="WTF")
+
+        response= self.client.get(MOVIE_URL, {"title": "Test"})
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        serializer_1 = MovieListSerializer(movie_1)
+        serializer_2 = MovieListSerializer(movie_2)
+        serializer_3 = MovieListSerializer(movie_3)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertNotIn(serializer_1.data, response.data)
+        self.assertIn(serializer_2.data, response.data)
+        self.assertNotIn(serializer_3.data, response.data)
+
+    def test_filter_movies_by_genre(self):
+        genre_1 = sample_genre(name="Action")
+        genre_2 = sample_genre(name="Horror")
+
+        movie = sample_movie()
+        movie_Action = sample_movie()
+        movie_Action_Horror = sample_movie()
+
+        movie_Action.genres.add(genre_1)
+        movie_Action_Horror.genres.add(genre_1, genre_2)
+
+        response = self.client.get(
+            MOVIE_URL,
+            {"genres": f"{genre_1.id},{genre_2.id}"}
+        )
+
+        serializer = MovieListSerializer(movie)
+        serializer_Action = MovieListSerializer(movie_Action)
+        serializer_Action_Horror = MovieListSerializer(movie_Action_Horror)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn(serializer_Action.data, response.data)
+        self.assertIn(serializer_Action_Horror.data, response.data)
+        self.assertNotIn(serializer.data, response.data)
```

Comment:
Code Quality Issue: This test case could be improved by also verifying that the correct movies are returned in the response when filtering by genre.